src/core/client/admin/test/community/community-rtl.spec.tsx

```diff
@@ -0,0 +1,545 @@
+  await createTestRenderer({
+    resolvers,
+  });
+
+  const userRow = await screen.findByTestId(`community-row-${user.id}`);
+  const changeRoleButton = within(userRow).getByLabelText("Change role");
+  changeRoleButton.click();
+
+  const popup = within(userRow).getByLabelText(
+    "A dropdown to change the user role"
+  );
+  const staffButton = await within(popup).findByRole("button", {
+    name: "Staff",
+  });
+  staffButton.click();
+
+  expect(resolvers.Mutation!.updateUserRole!.called).toBe(true);
+});
+
+it("can't change role as a moderator", async () => {
+  const moderator = users.moderators[0];
+  await createTestRenderer({
+    resolvers: createResolversStub<GQLResolver>({
+      Query: {
+        viewer: () => moderator,
+      },
+    }),
+  });
+
+  expect(screen.queryByLabelText("Change user status")).toBeInTheDocument();
+  expect(screen.queryByLabelText("Change role")).toBeNull();
+});
+
+it("change user role to Site Moderator and add sites to moderate", async () => {
+  const user = users.commenters[0];
+  const resolvers = createResolversStub<GQLResolver>({
+    Mutation: {
+      updateUserRole: ({ variables }) => {
+        expectAndFail(variables).toMatchObject({
+          userID: user.id,
+          role: GQLUSER_ROLE.MODERATOR,
+        });
+        const userRecord = pureMerge<typeof user>(user, {
+          role: variables.role,
+        });
+        return {
+          user: userRecord,
+        };
+      },
+      updateUserModerationScopes: ({ variables }) => {
+        expectAndFail(variables).toMatchObject({
+          moderationScopes: {
+            siteIDs: ["site-1"],
+          },
+          userID: user.id,
+        });
+        const userRecord = pureMerge<typeof user>(user, {
+          moderationScopes: {
+            scoped: true,
+            sites: [sites[0]],
+          },
+          role: GQLUSER_ROLE.MODERATOR,
+        });
+        return {
+          user: userRecord,
+        };
+      },
+    },
+    Query: {
+      settings: () => settingsWithMultisite,
+      sites: () => siteConnection,
+    },
+  });
+  await createTestRenderer({
+    resolvers,
+  });
+
+  const userRow = await screen.findByTestId(`community-row-${user.id}`);
+  const changeRoleButton = within(userRow).getByLabelText("Change role");
+  userEvent.click(changeRoleButton);
+
+  const popup = within(userRow).getByLabelText(
+    "A dropdown to change the user role"
+  );
+
+  const siteModButton = within(popup).getByTestId(
+    "role-change-siteModerator-button"
+  );
+  siteModButton.click();
+
+  const modal = await screen.findByTestId("site-role-modal");
+
+  // The submit button should be disabled until at least 1 site is selected
+  const submitButton = await screen.findByTestId(
+    "site-role-modal-submitButton"
+  );
+  expect(submitButton.classList.contains("Button-disabled")).toBe(true);
+
+  const siteSearchField = within(modal).getByTestId("site-search-textField");
+  userEvent.click(siteSearchField);
```

Comment:
You shouldn't need to click here, as `userEvent.type` by default clicks the element before typing.

Reply:
👍